mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -1029,8 +1034,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
       .persist(finalRDDStorageLevel)
     if (finalRDDStorageLevel != StorageLevel.NONE) {
       userIdAndFactors.count()
-      itemFactors.unpersist()
       itemIdAndFactors.count()
+      itemFactors.unpersist()
```

Comment:
`itemIdAndFactors.count` uses `itemFactors`. `itemFactors` should be unpersist after the action.